[12:28:17] [DEBUG] Running client sanity checks...
[12:28:19] [DEBUG] No setup function found in play
[12:28:19] [DEBUG] Found extension class Music in play: Attempting to load
[12:28:19] [DEBUG] Added callback: Music._menu
[12:28:19] [DEBUG] Added callback: Music._pause
[12:28:19] [DEBUG] Added callback: Music._resume
[12:28:19] [DEBUG] Added callback: Music._skip
[12:28:19] [DEBUG] Added callback: Music._stop
[12:28:19] [DEBUG] Added callback: Music.play
[12:28:19] [DEBUG] Loaded Extension: play
[12:28:20] [DEBUG] No setup function found in askbard
[12:28:20] [DEBUG] Found extension class Bard in askbard: Attempting to load
[12:28:20] [DEBUG] Added callback: Bard._askbard
[12:28:20] [DEBUG] Added callback: Bard._endbard
[12:28:20] [DEBUG] Added callback: Bard.set_bard_channel
[12:28:20] [DEBUG] Loaded Extension: askbard
[12:28:20] [DEBUG] No setup function found in noi_chu
[12:28:20] [DEBUG] Found extension class NoiChu in noi_chu: Attempting to load
[12:28:20] [DEBUG] Added callback: NoiChu._reset
[12:28:20] [DEBUG] Added callback: NoiChu._world_setup
[12:28:20] [DEBUG] Loaded Extension: noi_chu
[12:28:20] [DEBUG] No setup function found in db_refesh
[12:28:20] [DEBUG] Found extension class Database in db_refesh: Attempting to load
[12:28:20] [DEBUG] Loaded Extension: db_refesh
[12:28:20] [DEBUG] No setup function found in help
[12:28:20] [DEBUG] Found extension class Help in help: Attempting to load
[12:28:20] [DEBUG] Added callback: Help.help
[12:28:20] [DEBUG] Loaded Extension: help
[12:28:20] [DEBUG] Added callback: __join
[12:28:20] [DEBUG] Added callback: __leave
[12:28:20] [DEBUG] Added callback: _about
[12:28:20] [DEBUG] Added callback: _refresh
[12:28:20] [DEBUG] Added callback: _setup
[12:28:20] [DEBUG] Added callback: on_start
[12:28:20] [DEBUG] 6 callbacks have been loaded from __main__.
[12:28:20] [DEBUG] Added callback: _disconnect
[12:28:20] [DEBUG] Added callback: _on_websocket_ready
[12:28:20] [DEBUG] Added callback: on_autocomplete_completion
[12:28:20] [DEBUG] Added callback: on_autocomplete_error
[12:28:20] [DEBUG] Added callback: on_command_completion
[12:28:20] [DEBUG] Added callback: on_command_error
[12:28:20] [DEBUG] Added callback: on_component_completion
[12:28:20] [DEBUG] Added callback: on_component_error
[12:28:20] [DEBUG] Added callback: on_error
[12:28:20] [DEBUG] Added callback: on_modal_completion
[12:28:20] [DEBUG] Added callback: on_modal_error
[12:28:20] [DEBUG] Added callback: on_resume
[12:28:20] [DEBUG] 12 callbacks have been loaded from Client.
[12:28:20] [DEBUG] Attempting to login
[12:28:20] [DEBUG] Caching ingested rate limit data for: 78bb8553d9352a5a2f89f9def401287a
[12:28:20] [DEBUG] GET /users/@me Has exhausted its ratelimit (1000)! Locking route for 0.001 seconds
[12:28:20] [DEBUG] GET /users/@me Received 200 :: [0/1000 calls remaining]
[12:28:21] [DEBUG] Caching ingested rate limit data for: d28a133af187e91d71d96a223467ce49
[12:28:21] [DEBUG] GET /oauth2/applications/@me Has exhausted its ratelimit (1000)! Locking route for 0.001 seconds
[12:28:21] [DEBUG] GET /oauth2/applications/@me Received 200 :: [0/1000 calls remaining]
[12:28:21] [DEBUG] Caching ingested rate limit data for: 0a6d458104b103c7be293cc9b284d9ee
[12:28:21] [DEBUG] GET /gateway Received 200 :: [99/100 calls remaining]
[12:28:21] [INFO] Shard ID 0 | Starting Shard
[12:28:21] [INFO] Shard ID 0 | Shard is attempting to connect to gateway...
[12:28:21] [DEBUG] Shard ID 0 | Identification payload sent to gateway, requesting intents: 3407613
[12:28:22] [INFO] Shard ID 0 | Gateway connection established
[12:28:22] [DEBUG] Shard ID 0 | Session ID: dcd51baac39b17b6180d495e62525e51 Trace: ['["gateway-prd-us-east1-b-j82w",{"micros":163698,"calls":["id_created",{"micros":1346,"calls":[]},"session_lookup_time",{"micros":2210,"calls":[]},"session_lookup_finished",{"micros":18,"calls":[]},"discord-sessions-prd-2-193",{"micros":159644,"calls":["start_session",{"micros":121782,"calls":["discord-api-948796595-ztdbz",{"micros":106105,"calls":["get_user",{"micros":26046},"get_guilds",{"micros":4857},"send_scheduled_deletion_message",{"micros":11},"guild_join_requests",{"micros":2735},"authorized_ip_coro",{"micros":14}]}]},"starting_guild_connect",{"micros":62,"calls":[]},"presence_started",{"micros":280,"calls":[]},"guilds_started",{"micros":183,"calls":[]},"guilds_connect",{"micros":2,"calls":[]},"presence_connect",{"micros":37300,"calls":[]},"connect_finished",{"micros":37307,"calls":[]},"build_ready",{"micros":25,"calls":[]},"clean_ready",{"micros":1,"calls":[]},"optimize_ready",{"micros":1,"calls":[]},"split_ready",{"micros":0,"calls":[]}]}]}]']
[12:28:22] [DEBUG] Dispatching Event: websocket_ready
[12:28:22] [DEBUG] Dispatching Event: guild_join
[12:28:22] [DEBUG] Dispatching Event: guild_join
[12:28:22] [DEBUG] Dispatching Event: guild_join
[12:28:22] [DEBUG] Dispatching Event: guild_join
[12:28:22] [DEBUG] Dispatching Event: guild_join
[12:28:22] [DEBUG] Caching ingested rate limit data for: dd84d9818020c27975c0d5ab4a794d44
[12:28:22] [DEBUG] GET /applications/{application_id}/commands Has exhausted its ratelimit (1000)! Locking route for 0.001 seconds
[12:28:22] [DEBUG] GET /applications/{application_id}/commands Received 200 :: [0/1000 calls remaining]
[12:28:22] [DEBUG] Waiting for bucket dd84d9818020c27975c0d5ab4a794d44 to unlock.
[12:28:22] [DEBUG] Caching ingested rate limit data for: dd84d9818020c27975c0d5ab4a794d44
[12:28:22] [DEBUG] GET /applications/{application_id}/commands Has exhausted its ratelimit (1000)! Locking route for 0.001 seconds
[12:28:22] [DEBUG] GET /applications/{application_id}/commands Received 200 :: [0/1000 calls remaining]
[12:28:22] [INFO] Overwriting 0 with 15 application commands
[12:28:23] [DEBUG] Caching ingested rate limit data for: 3e32a240d3716487d7d8f6dae6c54f6b
[12:28:23] [DEBUG] PUT /applications/{app_id}/commands Received 200 :: [1/2 calls remaining]
[12:28:23] [DEBUG] Sync of 1 scopes took 1.5235304000016185 seconds
[12:28:23] [DEBUG] Dispatching Event: startup
[12:28:23] [DEBUG] Sending data to websocket: {"op": 3, "d": {"since": 1702877303653, "activities": [{"name": "l\u1ec7nh /help \u0111\u1ec3 gi\u00fap \u0111\u1ee1", "type": 0}], "status": "idle", "afk": false}}
[12:28:25] [DEBUG] 1174695146214928500::play should be called
[12:28:25] [DEBUG] Sending heartbeat every 41.25 seconds
[12:28:25] [DEBUG] Sending data to websocket: {"op": 1, "d": 7}
[12:28:25] [DEBUG] Shard ID 0 | ❤ Gateway is sending a Heartbeat
[12:28:26] [DEBUG] Dispatching Event: message_create
[12:28:26] [DEBUG] Caching ingested rate limit data for: 432f6223265d156004ac81366c5a948f
[12:28:26] [DEBUG] POST /interactions/{interaction_id}/aW50ZXJhY3Rpb246MTE4NjE3ODEzNDA0Nzk4OTg3MjpGU1BLOVJ0eVlPME1aTE5Bc1hpdE5LelNMU010RHZNcVBkSkgyckR6aHRMZHhKUzM3ZUk4MW1xWW0zdkFVZFhNMklCUzhzMERwaUxidXozV2VYeTNzS1VGTE9kYWw5bVg1ZDBqY3o1NnhITHdnNVMxYkZrNVlES0xvU3RweFZjUw/callback Received 204 :: [4/5 calls remaining]
[12:28:26] [DEBUG] Sending data to websocket: {"op": 4, "d": {"guild_id": 1067024293709496340, "channel_id": 1184720018127196202, "self_mute": false, "self_deaf": false}}
[12:28:26] [DEBUG] Waiting for voice connection data...
[12:28:26] [DEBUG] Shard ID 0 | ❤ Received heartbeat acknowledgement from gateway
[12:28:26] [DEBUG] Dispatching Event: voice_user_join
[12:28:26] [DEBUG] Attempting to initialise voice gateway...
[12:28:26] [DEBUG] Voice Connection has identified itself to Voice Gateway
[12:28:26] [DEBUG] Discord send VC Ready! Establishing a socket connection...
[12:28:26] [DEBUG] IP Discovery in progress...
[12:28:26] [DEBUG] Voice Initial Response Received: b'\x00\x02\x00F\x00\x00\xa2\xfd116.100.183.123\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xc2\x1f'
[12:28:26] [DEBUG] IP Discovered: 116.100.183.123 #49695
[12:28:26] [DEBUG] Sending data to websocket: {"op": 1, "d": {"protocol": "udp", "data": {"address": "116.100.183.123", "port": 49695, "mode": "xsalsa20_poly1305_lite"}}}
[12:28:26] [DEBUG] Starting UDP Keep Alive
[12:28:26] [DEBUG] Unhandled OPCODE: 18 = data = {'user_id': '610841389164396565', 'flags': 2}
[12:28:26] [DEBUG] Unhandled OPCODE: 20 = data = {'user_id': '610841389164396565', 'platform': 0}
[12:28:27] [INFO] Voice connection established; using xsalsa20_poly1305_lite
[12:28:27] [DEBUG] Sending heartbeat every 13.75 seconds
[12:28:27] [DEBUG] Sending data to websocket: {"op": 3, "d": 0.4310528591385918}
[12:28:27] [DEBUG] ❤ Voice Connection is sending Heartbeat
[12:28:27] [DEBUG] ❤ Heartbeat acknowledged after 0.04706 seconds
[12:28:30] [DEBUG] Dispatching Event: command_error
[12:28:30] [DEBUG] Dispatching Event: command_completion
[12:28:30] [DEBUG] Dispatching Event: error
[12:28:30] [INFO] Command Called: play with event.ctx.args = ['nụ cươi 9'] | event.ctx.kwargs = {'song': 'nụ cươi 9'}
[12:28:30] [ERROR] Ignoring exception in cmd `/play`:
Traceback (most recent call last):
  File "C:\Users\huuti\AppData\Local\Programs\Python\Python311\Lib\site-packages\interactions\client\client.py", line 1900, in __dispatch_interaction
    response = await callback
               ^^^^^^^^^^^^^^
  File "C:\Users\huuti\AppData\Local\Programs\Python\Python311\Lib\site-packages\interactions\client\client.py", line 1771, in _run_slash_command
    return await command(ctx, **ctx.kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\huuti\AppData\Local\Programs\Python\Python311\Lib\site-packages\interactions\models\internal\command.py", line 132, in __call__
    await self.call_callback(self.callback, context)
  File "C:\Users\huuti\AppData\Local\Programs\Python\Python311\Lib\site-packages\interactions\models\internal\application_commands.py", line 833, in call_callback
    return await self.call_with_binding(callback, ctx, *new_args, **new_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\huuti\AppData\Local\Programs\Python\Python311\Lib\site-packages\interactions\models\internal\callback.py", line 43, in call_with_binding
    return await callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Project\Python\qpneruy-Git\music\play.py", line 112, in play
    audio = YTDownloader.create_new_cls(audio_d)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Project\Python\qpneruy-Git\modules\YT_Downloader.py", line 107, in create_new_cls
    new_cls = cls(entry_data['url'])
                  ~~~~~~~~~~^^^^^^^
KeyError: 'url'

[12:28:41] [DEBUG] Sending data to websocket: {"op": 3, "d": 0.22833270643190573}
[12:28:41] [DEBUG] ❤ Voice Connection is sending Heartbeat
[12:28:41] [DEBUG] ❤ Heartbeat acknowledged after 0.07295 seconds
[12:28:55] [DEBUG] Sending data to websocket: {"op": 3, "d": 0.550549246849378}
[12:28:55] [DEBUG] ❤ Voice Connection is sending Heartbeat
[12:28:55] [DEBUG] ❤ Heartbeat acknowledged after 0.07415 seconds
[12:29:07] [DEBUG] Sending data to websocket: {"op": 1, "d": 10}
[12:29:07] [DEBUG] Shard ID 0 | ❤ Gateway is sending a Heartbeat
[12:29:07] [DEBUG] Shard ID 0 | ❤ Received heartbeat acknowledgement from gateway
[12:29:08] [DEBUG] Sending data to websocket: {"op": 3, "d": 0.2199095163178697}
[12:29:08] [DEBUG] ❤ Voice Connection is sending Heartbeat
[12:29:08] [DEBUG] ❤ Heartbeat acknowledged after 0.08011 seconds
[12:29:22] [DEBUG] Sending data to websocket: {"op": 3, "d": 0.04550659974017268}
[12:29:22] [DEBUG] ❤ Voice Connection is sending Heartbeat
[12:29:22] [DEBUG] ❤ Heartbeat acknowledged after 0.08059 seconds
[12:29:36] [DEBUG] Sending data to websocket: {"op": 3, "d": 0.6560076077976772}
[12:29:36] [DEBUG] ❤ Voice Connection is sending Heartbeat
[12:29:36] [DEBUG] ❤ Heartbeat acknowledged after 0.07314 seconds
[12:29:41] [DEBUG] Dispatching Event: disconnect
[12:29:41] [DEBUG] Stopping the bot.
[12:29:41] [INFO] Shard ID 0 | Stopping Shard
